ext/LinearSolveForwardDiff.jl

```diff
@@ -0,0 +1,80 @@
+module LinearSolveForwardDiff
+
+using LinearSolve
+isdefined(Base, :get_extension) ? 
+    (import ForwardDiff; using ForwardDiff: Dual) : 
+    (import ..ForwardDiff; using ..ForwardDiff: Dual)
+
+function _solve!(cache, alg, dAs, dbs, A, b, T; kwargs...)
+    @assert !(eltype(first(dAs)) isa Dual)
+    @assert !(eltype(first(dbs)) isa Dual)
+    @assert !(eltype(A) isa Dual)
+    @assert !(eltype(b) isa Dual)
+    reltol = cache.reltol isa Dual ? ForwardDiff.value(cache.reltol) : cache.reltol
+    abstol = cache.abstol isa Dual ? ForwardDiff.value(cache.abstol) : cache.abstol
+    u = eltype(cache.u) <: Dual ? ForwardDiff.value.(cache.u) : cache.u
+    cacheval = eltype(cache.cacheval.factors) <: Dual ? begin 
+        LinearSolve.LinearAlgebra.LU(ForwardDiff.value.(cache.cacheval.factors), cache.cacheval.ipiv, cache.cacheval.info)
+    end : cache.cacheval
+    cache2 = remake(cache; A, b, u, reltol, abstol, cacheval)
+    res = LinearSolve.solve!(cache2, alg, kwargs...) |> deepcopy
+    dresus = reduce(hcat, map(dAs, dbs) do dA, db
+        cache2.b = db - dA * res.u
+        dres = LinearSolve.solve!(cache2, alg, kwargs...)
+        deepcopy(dres.u)
+    end)
```

Comment:
Needing to `deepcopy` the results of the solves as they are being overwritten by subsequent solves using the same cache.

Reply:
I think if you hook into init and do a single batched solve then this is handled.

Reply:
Is there any documentation on how to do batched solves? I am unable to find how to do this anywhere. The possi bly closest thing I could find was https://discourse.julialang.org/t/batched-lu-solves-or-factorizations-with-sparse-matrices/106019/2 -- however, couldn't find the right function call.

Reply:
It's just `A\B` matrix instead of `A\b` vector

Reply:
I am not entirely sure what you mean in the context of LinearSolve.jl.

```julia
n = 4
A = rand(n, n)
B = rand(n, n)

A \ B  # works

mapreduce(hcat, eachcol(B)) do b
    A \ b
end # works

mapreduce(hcat, eachcol(B)) do b
    prob = LinearProblem(A, b)
    sol = solve(prob)
    sol.u
end # works

begin
    prob = LinearProblem(A, B)
    sol = solve(prob)
    sol.u
end # errors

```

Reply:
@avik-pal I thought you handled something with this?

Reply:
@avik-pal A ping on this. Is there another way to do this if we do not yet have batch dispatch?

Reply:
not for this case, but a case where `A` and `b` are both batched. Here you will have to see how Base handles it, there are special LAPACK routines for these